### PR TITLE
feat: CE-936 Add privacy indication field to complaint

### PR DESCRIPTION
**Pull request by marqueone-ps**
# Description
Adds a new checkbox field to the create and edit complaint forms. This new value also needs to be made available in the exported complaint.

Story # CE-936

# Acceptance Criteria
- [x] update model: 
   - [x] drop privacy_request_ind 
   - [x] add privacy_request
- [x] update interfaces to add new privacy property
- [x] update redux to use new privacy property
- [x] Add privacy request dropdown to create complaint
- [x] Add privacy request dropdown to edit complaint
- [x] make new privacy filed available for only ceeb
- [x] update export complaint template to use new field

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-647-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-647-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/v1/complaint/complaint.service.ts [hunk 1]
```diff
@@ -801,6 +801,7 @@ export class ComplaintService {
       builder.where("complaint.complaint_identifier = :id", { id });
       const result = await builder.getOne();
 
+      const test = 0;
```
Comment (afwilcox) [inline backend/src/v1/complaint/complaint.service.ts:804]: Not sure why Sonar didn't flag this, but this should not be here.

File: frontend/src/app/components/containers/complaints/details/caller-information.tsx [hunk 2]
```diff
@@ -3,18 +3,33 @@ import { useAppSelector } from "../../../../hooks/hooks";
...
+  const { name, primaryPhone, secondaryPhone, alternatePhone, address, email, reportedByCode, privacyRequest } =
+    useAppSelector(selectComplaintCallerInformation);
+  const enablePrivacyFeature = useAppSelector(isFeatureActive(FEATURE_TYPES.PRIV_REQ));
+
+  let privacy = "";
+  if (privacyRequest === "Y") {
+    privacy = "Yes";
+  } else if (privacyRequest === "N") {
+    privacy = "No";
+  }
```
Comment (afwilcox) [inline frontend/src/app/components/containers/complaints/details/caller-information.tsx:19]: Isn't this a duplication of something already being done in the automapper?

edit: and also again in the Reducer...

File: migrations/migrations/R__0.28.0_insert-privacy-feature.sql
Comment (afwilcox) [inline migrations/migrations/R__0.28.0_insert-privacy-feature.sql:1]: These won't get picked up by the Redux cache busting.

As per https://github.com/bcgov/nr-compliance-enforcement/wiki/Flyway-Database-Migrations this should go in  R__Create-Test-Data.sql

File: migrations/migrations/V0.28.0__CE-936.sql
Comment (afwilcox) [inline migrations/migrations/V0.28.0__CE-936.sql:1]: Please update to follow what was specified in the technical design which follows the NRS standards: https://apps.nrs.gov.bc.ca/int/confluence/display/NRMDDDG/Indicator+and+Flag+Column+Design+Patterns?#:~:text=with%20the%20values.-,Pattern%20%232%20%2D%C2%A0Use%20Column%20Check%20Constraint,-Use%20a%20check

- Column name should be prefixed with IS
- Check Constraint should be added to column

**Review by afwilcox — CHANGES_REQUESTED** [review]
No functional issues but please see comments on specific code items.

File: backend/src/v1/complaint/complaint.service.ts (hunk 1 above)
Comment (marqueone-ps) [reply backend/src/v1/complaint/complaint.service.ts:804]: I am equally surprised by that as well

File: frontend/src/app/components/containers/complaints/details/caller-information.tsx (hunk 2 above)
Comment (marqueone-ps) [reply frontend/src/app/components/containers/complaints/details/caller-information.tsx:19]: No, this isn't duplicating anything from the automapper. Whats happening the automapper is I'm creating a privacy property to handle our carbone workaround without messing up the complaint

Comment (afwilcox) [reply frontend/src/app/components/containers/complaints/details/caller-information.tsx:19]: Understood, 3 places for 3 different things:
1. Report Generation
2. Edit Screen drop downs
3. View Screen display

File: migrations/migrations/R__0.28.0_insert-privacy-feature.sql
Comment (marqueone-ps) [reply migrations/migrations/R__0.28.0_insert-privacy-feature.sql:1]: dropped the insert script for the features and moved them into the test-data script

File: migrations/migrations/V0.28.0__CE-936.sql
Comment (marqueone-ps) [reply migrations/migrations/V0.28.0__CE-936.sql:1]: updated the alter script to bring the privacy flag inline with NRS standards, and added a check constraint

**Review by afwilcox — CHANGES_REQUESTED** [review]
Changes all look good but there is something not working on the display (and maybe the export I didn't test that)

**Comment by marqueone-ps** [comment]
> Changes all look good but there is something not working on the display (and maybe the export I didn't test that)

export is working, but the thing that was off was the when I changed the name of the property vscode didn't change it everywhere, so it wasn't showing up in the ux. I've updated the interfaces to force this change back to working again.

**Review by afwilcox — APPROVED** [review]

**Pull request merged**